catroid/src/main/java/org/catrobat/catroid/utils/IconsUtil.java

```diff
@@ -102,6 +106,25 @@ public static void mapIconSizesToDeviceSize() {
 		largeIconBottomBar = (int) (metrics.ydpi * (LARGE_ICON_SIZE_BOTTOM_BAR_IN_CM * INCH_TO_CM));
 	}
 
+	public static void addIconToLabel(ViewGroup parent, Brick brick, Sprite sprite, Context context) {
+		for (int i = 0; i < parent.getChildCount(); i++) {
+			View child = parent.getChildAt(i);
+			if (child instanceof LinearLayout || child instanceof RelativeLayout || child instanceof BrickLayout) {
+				addIconToLabel((ViewGroup) child, brick, sprite, context);
+			}
+			if (child instanceof TextView) {
+				TextView textView = (TextView) child;
+				CategoryBricksFactory categoryBricksFactory = new CategoryBricksFactory();
+
+				String id = context.getResources().getResourceEntryName(textView.getId());
+				String subString = id.subSequence(id.length() - LABEL_KEYWORD.length(), id.length()).toString();
```

Comment:
This is potentially subject to IndexOutOfBoundsException if someone introduces a TextView with a really short id. Please add a check whether id size > LABEL_KEYWORD.length() and ignore these entries.

Reply:
or can this be applied to each BrickTextView?
This would make it easier since you could just check for instance of BrickTextView and get rid of dealing with substrings.

Reply:
Not to each because some bricks have multiple BrickTextViews. But it just dawned on me that I can simply only apply it to the first BrickTextView in every Brick, instead of applying it to the one with a "_label" keyword. I did that, should be the safer option.